### ISSUE TITLE
Browse Tables from DB Schema

### ISSUE THREAD

**Issue by oryxius** [sqlitebrowser/sqlitebrowser#1943]
Describe the new feature
--------------------------
I have many tables in when I need to move between these tables I have to open the tables list in the "Browse Data" window and navigate to my desired table -- or I would have to go to the "Database Structures" tab, find my desired table, right-click on it and choose "Browse Table" from the context menu. It would be faster and more productive if I were able to open the table just by double-clicking on its name in the list of tables in the DB Schema Window available to the right of the main "Browse Data" window, or at least open it through a right-click context menu option there. I hope you would be able to add this great feature soon, and thank you very much for a wonderful product!

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.
- No

#### Does this feature exist in another product or project? Please provide a link.
- I am not sure, but I would expect it.

#### Do you have a screenshot? Please add screenshots to help explain your idea.
- No.

**Comment by MKleusberg** [comment]
Oh, I wasn't even aware that this doesn't work! :wink: I have just added a "Browse Table" item to the right-click menu. So with the next nightly build it should work as expected. Are you ok to download and test that (should be available tomorrow from [here](https://nightlies.sqlitebrowser.org/latest/)? Double clicking still doesn't change the table though because it would shadow the collapse/uncollapse actions of the tree and I'm not sure whether that would be confusing for new users.

**Comment by oryxius** [comment]
Dear Martin,

Thank you so much for taking the time to look into this and to update the
program. Yes, it is now available in the test version. When do you expect
the next release?

All the best,

Muhamed

**Comment by MKleusberg** [comment]
Awesome! I'm glad it's working :smile: 

I'm going to close this issue then. As far as I know we don't have an expected date for the next release yet. But maybe by the end of the year? @justinclift What do you think?

**Comment by justinclift** [comment]
> As far as I know we don't have an expected date for the next release yet. But maybe by the end of the year?

@MKleusberg Been thinking about this.  Is there anything we're particularly waiting on?  If not, maybe lets set an ETA for creating the 3.12.x branch 1 month from now (eg mid-Sept), and get the release done after an translation + stabilising (eg beta release + fixing bugs that crop up).

@mgrojo @chrisjlocke @scottfurry @karim Thoughts?

**Comment by mgrojo** [comment]
> > As far as I know we don't have an expected date for the next release yet. But maybe by the end of the year?
> 
> @MKleusberg Been thinking about this. Is there anything we're particularly waiting on? If not, maybe lets set an ETA for creating the 3.12.x branch 1 month from now (eg mid-Sept), and get the release done after an translation + stabilising (eg beta release + fixing bugs that crop up).
> 
> @mgrojo @chrisjlocke @scottfurry @karim Thoughts?

Looking at [the closed issue list in milestone 7.12](https://github.com/sqlitebrowser/sqlitebrowser/milestone/15?closed=1), there are a lot of implemented enhancements and fixed bugs since the last release, so it might make sense to start thinking in the next release.

**Comment by MKleusberg** [comment]
@justinclift Mid-september sounds reasonable :smile: I think we should maybe have another look at issue #1701 and similar issues. As far as I remember this problem was introduced by some multithreading-related changes. There is also a TODO comment in the code [here](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/MainWindow.cpp#L196). This seems like the most important release blocker to me.

And :tada: for 3000 commits :wink:

**Comment by mgrojo** [comment]
> @justinclift Mid-september sounds reasonable smile I think we should maybe have another look at issue #1701 and similar issues. As far as I remember this problem was introduced by some multithreading-related changes. There is also a TODO comment in the code [here](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/MainWindow.cpp#L196). This seems like the most important release blocker to me.

Oh, I knew there was an issue for that, but I did not find it yesterday. I think I fixed that in the commit Nº 3000 aae2d4dea596072459f3e341689a9d65abc3c557 :smile: I took a look yesterday, precisely because I too considered it a release blocker. Being honest, I don't know exactly what was happening, but something related with the different signals that were triggered in the project loading code. @MKleusberg, you could take a look to the change just in case there is a better solution or at least to know what was happening out of curiosity.

**Comment by mgrojo** [comment]
Ah, I see that you made a change in cf631a02229d999c6379b29380028133240c0f7e for #1701 too. It seems that you went to the root of the problem. With that, I've checked that the call to `qApp->processEvents()` in https://github.com/sqlitebrowser/sqlitebrowser/blob/ed4c252dbb338207bb49578ef634477eb50854e4/src/MainWindow.cpp#L3054 is no longer needed. The remaining changes still make sense, don't they?

**Comment by MKleusberg** [comment]
@mgrojo Yes, I just tried to revert your entire commit and it's definitely broken again when doing so :smile: But the ```qApp->processEvents();``` probably can be removed safely. So from a project file point of view, your commit solves issue #1573, mine solves #1649. Good team effort then :smile:

**Comment by mgrojo** [comment]
Great! Perfect telepathic coordination :smile:

I just committed the removal of `qApp->processEvents()`.

**Comment by MKleusberg** [comment]
Awesome :smile: So what do you think? No more release blockers?

**Comment by mgrojo** [comment]
Nothing that I can think of.

**Comment by MKleusberg** [comment]
Cool, I guess we can start the release process any time then :smile:

**Comment by karim** [comment]
I *really* do like to have a shorter release cycle (1 to 3 months). Maybe we should open a new issue to discuss our release policy then write them down in the wiki.

We can keep the major releases (e.g. 3.11, 3.12) for new features, and minor releases (e.g. 3.11.1, 3.11.2) for confirmed bug fixes.

**Comment by justinclift** [comment]
> We can keep the major releases (e.g. 3.11, 3.12) for new features, and minor releases (e.g. 3.11.1, 3.11.2) for confirmed bug fixes.

Yeah, that pretty much sums up the approach we've been using.  3.12.0 will be the first release for the new series, with .1, .2, .3 (etc) as needed as we fix bugs in that specific branch. :smile:

**Comment by karim** [comment]
However, we only make a major release every 1 year or more. Then we quickly follow it up (within a couple of weeks) with a few (usually 1 or 2) minor releases to fix a couple of bugs that were reported by those who upgraded.

**Comment by justinclift** [comment]
Errr... yeah?
